src/textual/screen.py

```diff
@@ -621,6 +674,41 @@ def focus_previous(self, selector: str | type[QueryType] = "*") -> Widget | None
         """
         return self._move_focus(-1, selector)
 
+    def maximize(self, widget: Widget, container: bool = True) -> None:
+        """Maximize a widget, so it fills the screen.
+
+        Args:
+            widget: Widget to maximize.
+            container: Maximize container if possible.
+        """
+        if container:
+            # If we want to maximize the container, look up the dom to find a suitable widget
+            for maximize_widget in widget.ancestors:
```

Comment:
If I pass a container into this method, won't this code mean it's going to maximise the first widget with `allow_maximize = True` instead of the container I passed in?

Reply:
You don't pass in the container. You maximize a widget, and if that widget is within a maximizeable container, it will maximize that.

Feel free to suggest rewording to docstring to make that clearer.